services/filemanager/digitalOceanSpacesManager.go

```diff
@@ -133,8 +140,47 @@ func (manager *DOSpacesManager) ListFilesWithPrefix(prefix string, maxItems int6
 	return
 }
 
-func (manager *DOSpacesManager) DeleteObjects(locations []string) (err error) {
-	return
+func (manager *DOSpacesManager) DeleteObjects(keys []string) (err error) {
+	sess := manager.getSession()
+	if err != nil {
+		return fmt.Errorf(`error starting S3 session: %v`, err)
+	}
+
+	var objects []*s3.ObjectIdentifier
+	for _, key := range keys {
+		objects = append(objects, &s3.ObjectIdentifier{Key: aws.String(key)})
+	}
+
+	svc := s3.New(sess)
+
+	batchSize := 1000 // max accepted by DeleteObjects API
+	for i := 0; i < len(objects); i += batchSize {
+		j := i + batchSize
+		if j > len(objects) {
+			j = len(objects)
+		}
+		input := &s3.DeleteObjectsInput{
+			Bucket: aws.String(manager.Config.Bucket),
+			Delete: &s3.Delete{
+				Objects: objects[i:j],
+			},
+		}
+		_, err := svc.DeleteObjects(input)
+		if err != nil {
+			if aerr, ok := err.(awserr.Error); ok {
+				switch aerr.Code() {
+				default:
+					pkgLogger.Errorf(`Error while deleting S3 objects: %v, error code: %v`, aerr.Error(), aerr.Code())
```

Comment:
A lot of coded is added here, is it necessary for testing/refactoring ?

Reply:
this code is not added just for testing... for all the file managers, whichever function wasn't implemented, I implemented them. So, this is to delete from bucket.

Reply:
also, since DeleteObjects takes a max of 1000 objects at a time. So, we have the `for` loop to delete in batch of 1000 objects at a time.